### ISSUE TITLE
Un encadrant en formation peut devenir encadrant sur n'importe quelle collective s'il est encadrant dans une autre activité

### ISSUE THREAD

**Issue by fberu** [Club-Alpin-Annecy/collectives#496]
Use case:

- un adhérent A est encadrant Trail
- il est aussi encadrant en formation ski de rando

Lorsqu'un encadrant B crée une collective, il est possible d'ajouter n'importe quel encadrant quelque soit l'activité. Cet encadrant n'aura ensuite aucun droit d'édition de la collective. Par contre il ne doit pas être possible d'ajouter un encadrant en formation.

Le pb qui peut se poser avec le use case cité, c'est un encadrant B crée une collective de ski de rando. Comme l'adhérent A est déjà encadrant (Trail), l'encadrant B peut ajouter cet adhérent A comme encadrant de la collective alors qu'il est en formation.

Ce point dérange certains responsables d'activité de voir un encadrant en formation comme encadrant d'une collective.

A voir quel est le meilleur moyen:

- ne pouvoir ajouter comme encadrant que les encadrants de la ou les activités de la collective
- interdire de sélectionner un encadrant en formation de la ou les activités de la collective

**Comment by gdaviet** [comment]
Si je comprend bien:

- ne pouvoir ajouter comme encadrant que les encadrants de la ou les activités de la collective => C'est déjà le cas à moins d'être en mode "multi-activité". (mais j'ai déjà observé que cette  cache  se cochait parfois toute seule, peut être que la confusion vient de là)
 - interdire de sélectionner un encadrant en formation de la ou les activités de la collective => c'est déjà le cas, les encadrant en formation n'ont aucun droit particulier. Le comportement observé vient uniquement du fait qu'il soit encadrant d'une activité et que la collective est en mode "multi-activités"

**Comment by shaiton** [comment]
C’est probablement le cas que j’ai eu :
- collective « formation »
- activitées (optionnel) « alpinisme ». Ça sonne bien, mais l’activitée optionnelle m’est proposée par défaut.
En tout cas j’imagine ne pouvoir ajouter que les encadrants correspondant à l’équipe d’encadrant des activités sélectionnées.

Par contre, si on a des encadrants pour le type « formations » c’est impossible à filtrer.
C’est surtout un problème d’utilisation :  
- pas de raison d’ajouter des encadrants inconnus
- j’ai le cas d’encadrants qui ne comprennent pas la différence entre « encadrant » (tous sont coencadrant à la sortie sauf le responsable) et « coencdrant en formation ». M’enfin...

Aussi peut-être qu’un moyen d’accès à la liste des encadrants/coencadrant en formation par activité pour tous les encadrants permettrait d’y voir plus clair ? Si c’est intéressant c’est forcément une nouvelle fonctionnalité…

**Comment by gdaviet** [comment]
Oui, les types d'événement pour lesquels l'activité est optionelle (comme Formations) fonctionne de la même façon que le multi-activité: i.e. on autorise à avoir des encadrants de tous types d'activité, tant qu'il y en un dans le lot qui est diplômé  pour les activités choisies (ca permet de faire une formation "s'orienter en ski de rando" avec un encadrant ski de rando et un encadrant carto, ou d'ajourer des bénévoles "soirée" sur les WEI, etc). 
Le procédé actuel est: "il faut ajouter un encadrant de l'activité A pour pour voir ajouter l'activité A à la collective"... du coup il faut forcément ajouter des encadrants "inconnus" à un moment pour pouvoir ajouter leurs activités.

Après, si c'est une problème de filtre dans l'autocomplete, peut être que trier les résultats avec les activités existantes pourrait aider ?  Et indiquer à côté des noms d'encadrants pour quelles activités ils sont habilités (et pas juste en formation)

Bonne idée d'avoir la liste des encadrants en formation dispo quelque part... à voir où
